bot/exts/backend/auto_config.py

```diff
@@ -0,0 +1,182 @@
+import os
+import typing as t
+import random
+
+from pathlib import Path
+
+import discord
+import yaml
+
+from discord.errors import Forbidden, NotFound
+from discord.ext import commands
+from discord.ext.commands.context import Context
+from loguru import logger
+
+
+class AutoConfig(commands.Cog):
+    """A cog responsible for constructing config files"""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        with open('bot/config/config-list.yaml') as f:
+            self.config_list = yaml.safe_load(f)
+            self.auto = self.config_list['Auto-config']
+            self.config_list.pop('Auto-config')
+
+    @property
+    def guild_id(self) -> int:
+        """ Check's if the guild exists or not, if not then raises NotFound."""
+
+        if not self.bot.get_guild(int(os.getenv('Guild_ID'))):
+            logger.info(f"Cannot find the guild with id {os.getenv('Guild_ID')}. Rasing error!")
+            raise NotFound(f"Cannot find guild with ids {os.getenv('Guild_ID')}")
+        return int(os.getenv('Guild_ID'))
+
+    def required(self) -> list:
+        """Returns the list of information that are required for the bot to run."""
+
+        with open('bot/config/default-config.yaml') as f:
+            required_config = yaml.safe_load(f)
+
+        for x, y in self.config_list.items():
+            for k, v in y.items():
+                if v['optional'] is True:
+                    try:
+                        del required_config[x][k]
+                    except KeyError:
+                        continue
+
+        return required_config
+
+    def check(self) -> t.Union[bool, dict]:
+        """ A check that checks if the required channels, roles etc. exist or not."""
+
+        guild = self.bot.get_guild(self.guild_id)
+
+        #checking to see if the server has the required roles or not
+        roles = [x for x in self.required()['Roles'].keys() if not discord.utils.get(guild.roles, name=x)]
+
+        #checking for the required channels
+        channels = [x for x in self.required()['Roles'].keys(
+        ) if not discord.utils.get(guild.channels, name=x)]
```

Comment:
```suggestion
        channels = [x for x in self.required()['Roles'].keys() 
        if not discord.utils.get(guild.channels, name=x)]
```
Keep the brackets on the same line.